9HW/3_task.js

```diff
@@ -0,0 +1,273 @@
+            return;
+            } else {
+                this.teamSize += num;
+            }
+        }
+ }
+ class Designer extends Employee {
+    designTools = [];
+    constructor(firstName, lastName, salary, designTools = []) {
+        super(firstName, lastName, salary);
+        for (const designTool of designTools) {
+            this.designTools.push(designTool);
+        }
+    };
+    addDesignTool(tool){
+        if (typeof tool !== "string") {
+        console.log('Некорректный тип данных');
+        return;
+        } else {
+            this.designTools.push(tool);
+        }
+    }
+ }
+ class Company {
+    #employees = [];
+    constructor(title, phone, address, employees = []) {
+        this.title = title;
+        this.phone = phone;
+        this.address = address;
+        this.salary;
+        for (const employee of employees) {
+            this.#employees.push(employee);
+        }
+    }
+    get title() {
+        return this._title;
+    }
+    get phone() {
+        return this._phone;
+    }
+    get address() {
+        return this._address;
+    }
+    set title(title) {
+        if (typeof title !== "string") {
+        console.log('Некорректный тип данных');
+        return;
+        }
+        this._title = title;
+    }
+    set phone(phone) {
+        if (typeof phone !== "string") {
+        console.log('Некорректный тип данных');
+        return;
+        }
+        this._phone = phone;
+    }
+    set address(address) {
+        if (typeof address !== "string") {
+        console.log('Некорректный тип данных');
+        return;
+        }
+        this._address = address;
+    }
+    addEmployee(employee) {
+        if (employee instanceof Employee) {
+            let index = this.getEmployees().findIndex(emp => emp.fullName() === (employee.firstName + employee.lastName));
+            if (index !== -1) {
+                console.log('Работник с такими ФИ уже есть!');
+                console.log(this.getEmployees().at(index))
+            } else {
+                this.#employees.push(employee);
+            }
+        }
+    }
+    getEmployees() {
+        return this.#employees;
+    }
+    getInfo() {
+        return ` Компания: ${this.title} \n Адрес: ${this.address} \n Количество сотрудников: ${this.#employees.length}`;
+    }
+    #getEmployeeIndex(firstName) {
+        if (typeof firstName !== "string") {
+            return -1;
+            }
+        let index = this.getEmployees().findIndex(emp => emp._firstName === firstName);
+        return index;
+    }
+    findEmployeeByName(firstName) {
+        let index = this.#getEmployeeIndex(firstName);
+        if (index != -1) {
+        return this.#employees[index];
+        } else {
+            console.log('Нет такого сотрудника')
+        }
+    }
+    removeEmployee(firstName) {
+        if (this.#getEmployeeIndex(firstName) != -1) {
+        this.getEmployees().splice(this.#getEmployeeIndex(firstName), 1);
```

Comment:
Один раз нашли и используем
```suggestion
       const index = this.#getEmployeeIndex(firstName);
        if (index != -1) {
        this.getEmployees().splice(index, 1);
```